File: insidentesTypescript/src/types/Incidentes/Observacion.ts
import {
  Entity, PrimaryGeneratedColumn, Column,
  ManyToOne, JoinColumn
} from "typeorm";
import { Usuario } from "../Comunidades/Usuario"; // Asegúrate de importar la clase Usuario correcta
import { Incidente } from "./Incidente";

@Entity()
export class Observacion {
  @PrimaryGeneratedColumn()
  id!: number;

  @ManyToOne(() => Usuario, { eager: true })
  @JoinColumn({ name: "usuario_id" })
  usuario: Usuario;
  // ojo que puede faltar un @ManyToOne que va a incidente
  incidente!: Incidente;

  @Column()
  observado: string;

  constructor(usuario: Usuario, observado: string) {
    this.usuario = usuario;
    this.observado = observado;
  }
}

export default Observacion;

File: src/main/java/ar/edu/utn/frba/dds/models/incidentes/Observacion.java
package ar.edu.utn.frba.dds.models.incidentes;

import ar.edu.utn.frba.dds.models.comunidades.Perfil;
import ar.edu.utn.frba.dds.models.comunidades.Usuario;
import ar.edu.utn.frba.dds.models.persistencia.Persistente;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Getter
@Setter
public class Observacion extends Persistente {

    @ManyToOne
    @JoinColumn(name = "usuario_id", referencedColumnName = "id")
    private Usuario usuario;

    @Column(name="observado")
    private String observado;

    public Observacion(Usuario usuario, String observado) {
        this.usuario = usuario;
        this.observado = observado;
    }

    public Observacion() {
    }
}
